src/main/java/io/ticktok/Listener/TickListener.java

```diff
@@ -10,23 +12,27 @@
 
     private final static String QUEUE_NAME = "ticktok-queue";
 
-    public static void listen(ClockChannel clockChannel, Runnable runnable) throws Exception {
-        Connection connection = createConnection(clockChannel);
-        Channel channel = connection.createChannel();
-        channel.queueDeclare(QUEUE_NAME, false, false, false, null);
-        System.out.println(" [*] Waiting for messages. To exit press CTRL+C");
+    public static void listen(ClockChannel clockChannel, Runnable runnable) throws TicktokException{
+        try {
+            Connection connection = createConnection(clockChannel);
+            Channel channel = connection.createChannel();
+            channel.queueDeclare(QUEUE_NAME, false, false, false, null);
+            System.out.println(" [*] Waiting for messages. To exit press CTRL+C");
 
-        Consumer consumer = new DefaultConsumer(channel) {
-            @Override
-            public void handleDelivery(String consumerTag, Envelope envelope,
-                                       AMQP.BasicProperties properties, byte[] body)
-                    throws IOException {
-                String message = new String(body, "UTF-8");
-                System.out.println(" [x] Received '" + message + "'");
-                runnable.run();
-            }
-        };
-        channel.basicConsume(QUEUE_NAME, true, consumer);
+            Consumer consumer = new DefaultConsumer(channel) {
+                @Override
+                public void handleDelivery(String consumerTag, Envelope envelope,
+                                           AMQP.BasicProperties properties, byte[] body)
+                        throws IOException {
+                    String message = new String(body, "UTF-8");
+                    System.out.println(" [x] Received '" + message + "'");
+                    runnable.run();
+                }
+            };
+            channel.basicConsume(QUEUE_NAME, true, consumer);
+        } catch (Exception e) {
```

Comment:
Broad exception

Reply:
why? I think we need to put a lot more thought into this, this client is almost done 
there will be no too many revisions for it

Reply:
We cannot fix this.
`factory.setUri(new URI(clockChannel.getUri()));` throws *URISyntaxException* that inherit directly from *Exception*

same story goes for `return factory.newConnection();` that throws *TimeoutException* that inherit directly from *Exception*